File: jApp/src/main/java/j/s/yarlykov/ui/ForecastActivity.java
package j.s.yarlykov.ui;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.Formatter;
import java.util.Locale;

import j.s.yarlykov.R;
import j.s.yarlykov.data.domain.CityForecast;

public class ForecastActivity extends AppCompatActivity {

    public static final String EXTRA_FORECAST = ForecastActivity.class.getSimpleName() + ".extra.FORECAST";

    private String NO_DATA;
    private CityForecast forecast;
    private TextView tvCity, tvTemperature, tvWind, tvHumidity, tvPressure;
    private ImageView ivSky;

    public static void start(Context context, CityForecast forecast) {
        Intent intent = new Intent(context, ForecastActivity.class);
        intent.putExtra(EXTRA_FORECAST, forecast);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_forecast);
        forecast = (CityForecast) getIntent().getSerializableExtra(EXTRA_FORECAST);
        initView();
    }

    @Override
    protected void onResume() {
        super.onResume();
        renderForecast();
    }

    /**
     * TODO: Find views
     */
    private void initView() {
        NO_DATA = getResources().getString(R.string.noData);
        ivSky = findViewById(R.id.iv_sky);
        tvCity = findViewById(R.id.tv_city);
        tvTemperature = findViewById(R.id.tv_temperature);
        tvWind = findViewById(R.id.tv_wind);
        tvHumidity = findViewById(R.id.tv_humidity);
        tvPressure = findViewById(R.id.tv_pressure);
    }

    /**
     * TODO: Отрисовать прогноз на экране
     */
    private void renderForecast() {

        // TODO: Weather image
        ivSky.setImageResource(forecast.getImgId());

        //TODO: City (Uppercase first letter)
        String city = forecast.getCity();
        tvCity.setText(city.substring(0, 1).toUpperCase() + city.substring(1).toLowerCase());

        // TODO: Temperature
        Formatter fmt = new Formatter();
        fmt.format("%+2d \u2103", forecast.getTemperature());
        tvTemperature.setText(fmt.toString());
        fmt.close();

        // TODO: Wind
        if(forecast.getWind() != CityForecast.EMPTY_VAL) {
            fmt = new Formatter();
            fmt.format("%2d %s", forecast.getWind(), getResources().getString(R.string.infoWind));
            tvWind.setText(fmt.toString());
            fmt.close();
        } else {
            tvWind.setText(NO_DATA);
        }

        // TODO: Humidity
        if(forecast.getHumidity() != CityForecast.EMPTY_VAL) {
            fmt = new Formatter();
            fmt.format("%2d %%", forecast.getHumidity());
            tvHumidity.setText(fmt.toString());
            fmt.close();
        } else {
            tvHumidity.setText(NO_DATA);
        }

        // TODO: Pressure
        Locale current = getResources().getConfiguration().locale;
        boolean isRu = current.getCountry() == "RU";
        if(forecast.getPressure(isRu) != CityForecast.EMPTY_VAL) {
            fmt = new Formatter();
            fmt.format("%4d %s", (int) forecast.getPressure(isRu), getResources().getString(R.string.infoPressure));
            tvPressure.setText(fmt.toString());
            fmt.close();
        } else {
            tvPressure.setText(NO_DATA);
        }
    }
}

File: kApp/src/main/java/k/s/yarlykov/ui/ForecastActivity.kt
package k.s.yarlykov.ui

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_forecast.*;
import k.s.yarlykov.R
import k.s.yarlykov.data.domain.CityForecast
import k.s.yarlykov.data.domain.Forecast.Companion.EMPTY_VAL
import k.s.yarlykov.util.Utils.isRu

class ForecastActivity : AppCompatActivity() {

    lateinit var NO_DATA: String
    lateinit var forecast: CityForecast

    companion object {
        private val EXTRA_FORECAST = ForecastActivity::class.java.simpleName + ".extra.FORECAST"

        fun start(context: Context, forecast: CityForecast) {
            val intent = Intent(context, ForecastActivity::class.java).apply {
                putExtra(EXTRA_FORECAST, forecast)
            }
            context.startActivity(intent)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_forecast)
        forecast = intent.getSerializableExtra(EXTRA_FORECAST) as CityForecast
        NO_DATA = getResources().getString(R.string.noData)
    }

    override fun onResume() {
        super.onResume()
        renderForecast()
    }

    fun renderForecast() {

        // Set Weather image
        ivSkyA.setImageResource(forecast.imgId)

        //Set City (Uppercase first letter)
        tvCityA.text = (forecast.city.capitalize())

        // Set Temperature
        tvTemperatureA.text = (String.format("%+2d \u2103", forecast.temperature))

        // Set Wind
        if (forecast.wind != EMPTY_VAL) {
            tvWindA.text = (String.format("%2d %s", forecast.wind, getResources().getString(R.string.infoWind)))
        } else tvWindA.text = (NO_DATA)

        // Set Humidity
        if (forecast.humidity != EMPTY_VAL.toInt()) {
            tvHumidityA.text = (String.format("%2d %%", forecast.humidity))
        } else tvHumidityA.text = (NO_DATA)

        // Set Pressure
        if (forecast.pressureMm != EMPTY_VAL.toInt()) {
            tvPressureA.text = (String.format("%4d %s", forecast.getPressure(isRu()), getResources().getString(R.string.infoPressure)))
        } else tvPressureA.text = (NO_DATA)
    }
}